providers/apis/kraken/api_handler.go

```diff
@@ -20,25 +19,16 @@ var _ types.PriceAPIDataHandler = (*APIHandler)(nil)
 // for more information about the Kraken API, refer to the following link:
 // https://docs.kraken.com/rest/
 type APIHandler struct {
-	// market is the config for the Kraken API.
-	market types.ProviderMarketMap
 	// api is the config for the Kraken API.
 	api config.APIConfig
+	// cache maintains the latest set of tickers seen by the handler.
+	cache types.ProviderTickers
```

Comment:
One thing im thinking while reading this PR:

Can we really call this a "cache" if there is no eviction mechanism ?  And do we need eviction?

Reply:
The cache is technically reset everytime the provider is so this should be safe, in terms of how the cache is actually going to operate if there are multiple of the same off-chain tickers is that the latest one will be stored which is fine. We do now have a uniqueness constraint on the off-chain-ticker tho.